frontend/public/components/graphs/query-browser.jsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as classNames from 'classnames';
+import { addTraces, relayout, restyle } from 'plotly.js/lib/core';
+
+import { connectToURLs, MonitoringRoutes } from '../../monitoring';
+import { Dropdown, ExternalLink, LoadingInline } from '../utils';
+import { formatPrometheusDuration, parsePrometheusDuration } from '../utils/datetime';
+import { Line_ } from './line';
+
+const spans = ['5m', '15m', '30m', '1h', '2h', '6h', '12h', '1d', '2d', '1w', '2w'];
+const dropdownItems = _.zipObject(spans, spans);
+
+class QueryBrowser_ extends Line_ {
```

Comment:
General comment that we've been trying to get away from inheritance in React components, but I'm fine with it here since it's how the other charts work.

https://reactjs.org/docs/composition-vs-inheritance.html